daemon/cluster/cluster.go

```diff
@@ -334,8 +334,9 @@ func (c *Cluster) Cleanup() {
 		c.mu.Unlock()
 		return
 	}
-	defer c.mu.Unlock()
 	state := c.currentNodeState()
+	c.mu.Unlock()
```

Comment:
I've updated the PR to avoid acquiring `c.mu` to read `c.nr` or call `c.currentNodeState` when `controlMutex` is already held. PTAL.